### PR TITLE
[PLAYER-5463] Add configuration option to disable pause on app backgrounding

### PR DESCRIPTION
**Pull request by oSulyma**
**Ticket goal:** Add configuration option to disable pause on app backgrounding
**How PR achieves the goal:**  Please see major description inside PR https://git.corp.ooyala.com/projects/PBA/repos/ios-sdk/pull-requests/506/overview for PLAYER-5463 in iOS-SDK. To enable background playback property of instance OOOptions was must be assigned to OOBackgroundPlaybackModeAllowed
**Affected repo-s:** Sample-Apps + ios-SDK (must be synchronized, switch for appropriate branch)
**Unit tests:** Unit test not added, because there is no test target for BasicPlayback-SA.
**SampleApp for testing**: BasicPlayback-SA (or any native SA). Please note: to enable background playback go to BasicSimplePlayerViewController -> -viewDidLoad: and change `options.backgroundMode` from `OOBackgroundPlaybackModeDenied` (that must be kept in repository as default) to `OOBackgroundPlaybackModeAllowed`

### CODE REVIEW

File: BasicPlaybackSampleApp/BasicPlaybackSampleApp/Players/BasicSimplePlayerViewController.m [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 #import "BasicSimplePlayerViewController.h"
 #import <OoyalaSDK/OoyalaSDK.h>
```
Comment (Artur-Sakhno) [inline BasicPlaybackSampleApp/BasicPlaybackSampleApp/Players/BasicSimplePlayerViewController.m:10]: We can use `@import OoyalaSDK;` here

File: BasicPlaybackSampleApp/BasicPlaybackSampleApp/Players/BasicSimplePlayerViewController.m [hunk 2]
```diff
@@ -8,6 +8,7 @@
...
+#import <OoyalaSDK/OOEnums.h>
```
Comment (Artur-Sakhno) [inline BasicPlaybackSampleApp/BasicPlaybackSampleApp/Players/BasicSimplePlayerViewController.m]: Same `@import OoyalaSDK.OOEnums;`

**Review by maxkupetskii — CHANGES_REQUESTED** [review]
I think we have to check how `ControlsLockSampleApp` works now because it contains some manual code for altering `OoyalaSDK` behavior. So it's possible that this manual code should be either removed or changed

File: BasicPlaybackSampleApp/BasicPlaybackSampleApp/Players/BasicSimplePlayerViewController.m (hunk 2 above)
Comment (oSulyma) [reply BasicPlaybackSampleApp/BasicPlaybackSampleApp/Players/BasicSimplePlayerViewController.m]: Done

File: BasicPlaybackSampleApp/BasicPlaybackSampleApp/Players/BasicSimplePlayerViewController.m (hunk 1 above)
Comment (oSulyma) [reply BasicPlaybackSampleApp/BasicPlaybackSampleApp/Players/BasicSimplePlayerViewController.m:10]: Done

**Review by maxkupetskii — APPROVED** [review]

**Review by cacel — APPROVED** [review]
LGTM

**Review by DmitriiMaslov — APPROVED** [review]

**Pull request merged**